tests/integration_tests/conftest.py

```diff
@@ -459,8 +468,27 @@ def _generate_profile_report() -> None:
     log.info(command, "final.stats")
 
 
+# https://docs.pytest.org/en/stable/reference/reference.html#pytest.hookspec.pytest_sessionstart
+def pytest_sessionstart(session) -> None:
+    """do session setup"""
+    global _SESSION_CLOUD
+    try:
+        _SESSION_CLOUD = get_session_cloud()
+        setup_image(_SESSION_CLOUD)
+    except Exception as e:
+        pytest.exit(
+            f"{type(e).__name__} in session setup: {str(e)}", returncode=2
+        )
+
+
 def pytest_sessionfinish(session, exitstatus) -> None:
+    """do session teardown"""
     if integration_settings.INCLUDE_COVERAGE:
         _generate_coverage_report()
     elif integration_settings.INCLUDE_PROFILE:
         _generate_profile_report()
+    try:
```

Comment:
I think it's possible the report generations above these lines could fail causing us to not cleanup. We should ensure the following lines get run regardless.

Reply:
Agreed, see the latest push.